File: financeiro-api/src/main/java/com/financeiro/financeiroapi/model/ParcelasContratos.java
package com.financeiro.financeiroapi.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.sql.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.Valid;

import com.sun.istack.NotNull;

import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Data
@Entity
@Table(name = "parcelas_contratos")
public class ParcelasContratos implements Serializable{
	private static final long serialVersionUID = 1L;

	@EqualsAndHashCode.Include
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Valid
	@NotNull
	@ManyToOne
	@JoinColumn(name = "contrato_id")
	private Contratos  contratos;
	
	private Long parcela;
	private BigDecimal valor ;
	private BigDecimal juros ;
	private BigDecimal multa ;
	private BigDecimal correcao ;
	private BigDecimal desconto ;
	private BigDecimal valorPago ;
	private Date dataVencimento ;
	private Date dataPagamento ;
	private String documento ;
	private Long bancoId ;
	private String observacao;
	
}

File: financeiro-ui/src/app/shared/models/parcelas-contratos.ts
import { Contratos } from './contratos';
import { DecimalPipe } from '@angular/common';
import { BaseResourceModel } from './base-resource.model';

export class ParcelasContratos extends BaseResourceModel {

   constructor(
      public id?: number,
      public parcela?: number,
      public valor?: DecimalPipe,
      public juros ?: DecimalPipe,
      public multa ?: DecimalPipe,
      public correcao ?: DecimalPipe,
      public desconto ?: DecimalPipe,
      public valorPago ?: DecimalPipe,
      public dataVencimento?: Date ,
      public dataPagamento?: Date,
      public documento?: string,
      public bancoId?: number,
      public observacao?: string,

      public contratoId?: number,
      public contratos?: Contratos,
   ) {
     super();
   }

   static fromJson(jsonData: any): ParcelasContratos {
     const parcelasContratos = {
       ...jsonData,
       contratosId: jsonData["contratos"]["id"],
     };
     return Object.assign(new ParcelasContratos(), parcelasContratos);
   }

   static toJson(obj: any): ParcelasContratos{
     delete obj['contratos'];
     return Object.assign(new ParcelasContratos(), obj);
   }
}
